src/main/scala/com/cognite/spark/datasource/DefaultSource.scala

```diff
@@ -158,5 +167,38 @@ class DefaultSource
       case _ => sys.error("Unknown resource type: " + resourceType)
     }
   }
+
+  override def createRelation(
+      sqlContext: SQLContext,
+      mode: SaveMode,
```

Comment:
Couldn't we use `SaveMode` instead of `.option("savemode", "our-thing")`?

Seems `SaveMode.Overwrite` maps well enough to upsert. `SaveMode.Append` may not make sense for our data. `SaveMode.Ignore` would be "don't do upsert on conflict, just leave it how it is" I guess. The default `SaveMode.ErrorIfExists` would be "don't do upsert and just abort on conflict".

Reply:
SaveMode works as well, but we can't define new `SaveMode`s (at least I couldn't find a way to do it). With `option()` we can get the exact names, which I think is clearer for the user.

Reply:
Maybe using `SaveMode` if "savemode" isn't defined? This should be documented, of course. :)